src/main/java/com/auctionapp/api/service/BidService.java

```diff
@@ -0,0 +1,99 @@
+package com.auctionapp.api.service;
+
+import java.sql.Timestamp;
+import java.time.Instant;
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+import com.auctionapp.api.model.dto.BidDto;
+import com.auctionapp.api.model.entities.Bid;
+import com.auctionapp.api.repository.BidRepository;
+
+import org.springframework.stereotype.Service;
+
+@Service
+public class BidService {
+
+    private final BidRepository bidRepository;
+	private final AuctionService auctionService;
+
+    public BidService(final BidRepository bidRepository,
+					  final AuctionService auctionService) {
+        this.bidRepository = bidRepository;
+		this.auctionService = auctionService;
+    }
+
+    public List<BidDto> getAuctionBids(final UUID auctionId) {
+        final List<Bid> bids = bidRepository.findAllByAuctionId(auctionId);
+		return bids.stream().map(t -> toPayload(t)).collect(Collectors.toList());
+    }
+
+	public BidDto save(final BidDto payload) {
+		if (existsAmongBidders(payload.getBuyer().getId(), payload.getAuction().getId())) {
+			return toPayload(update(payload));
+		}
+
+		payload.setBidDate(Timestamp.from(Instant.now()));
+        Bid bid = fromPayload(payload);
+        bid = bidRepository.save(bid);
+		bid.setAuction(auctionService.update(payload.getAuction(), payload.getBidAmount()));
+        return toPayload(bid);
+	}
+
+	public Bid update(final BidDto payload) {
+        Optional<Bid> bidEntity = bidRepository.findByBuyerIdAndAuctionId(payload.getBuyer().getId(), payload.getAuction().getId());
+
+        if (bidEntity.isPresent()) {
+			if (payload.getBidAmount() > bidEntity.get().getBidAmount()) {
+				bidEntity.get().setBidAmount(payload.getBidAmount());
+				bidEntity.get().setBidDate(Timestamp.from(Instant.now()));
+				Bid  bid = bidRepository.save(bidEntity.get());
+				bid.setAuction(auctionService.update(payload.getAuction(), payload.getBidAmount()));
+				return bid;
+			}
+			throw new RuntimeException("New bid amount has to be greater than the previous ones!");
+        }
+        throw new RuntimeException("Bid with id " + payload.getId() + " does not exist!");
+    }
+
+	public boolean validateBid(final BidDto bid){
+		return validateBidAmount(bid.getBidAmount(), bid.getAuction().getHighestBid()) 
+                && !isBidderEqualSeller(bid.getBuyer().getId(), bid.getAuction().getSeller().getId());
```

Comment:
You should also validate that the placed bid is higher than the starting price for the auction. What if the highest bid is null?

Reply:
Is it okay if I, during the creation of the auction record, set the highest bid to the start price value, so in that way the first bid must be higher than the start price?